lib/topics/index.js

```diff
@@ -57,6 +58,56 @@ Topics.prototype.getTopic = function(topicName, prefix, topicUid) {
 };
 
 /**
+ *
+ * Internal function to get Promises for done and error topics.
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {*}
+ */
+Topics.prototype.getTopicPromises = function getTopicPromises(doneTopicPromise, errorTopicPromise) {
+  var deferred = q.defer();
+
+  doneTopicPromise.then(function(createdItem) {
+    deferred.resolve(createdItem);
+  });
+
+  errorTopicPromise.then(function(error) {
+    deferred.reject(error);
+  });
+
+  return deferred.promise;
+};
+
+/**
+ *
+ * Internal function to get Promises for done and error topics.
+ *
+ * @param {Object} config - specific config settings (if any)
+ * @param {Object} topicDefaults - topic-specific values for DONE_PREFIX, ERROR_PREFIX, and TOPIC_TIMEOUT
+ * @param {MediatorTopicUtility} topicGenerator
+ * @param {string} topicName   - The name of the topic to generate
+ * @param {string} [topicUid]  - A topic UID if required.
+ * @returns {Promise} - A promise for the topic.
+ */
+Topics.prototype.getErrorAndDoneTopicPromises = function getErrorAndDoneTopicPromises(config, topicDefaults, topicGenerator, topicName, topicUid) {
```

Comment:
I don't think you need `topicGenerator` here. The function you are in is the generator.